internal/tui/components/inputbox/inputbox.go

```diff
@@ -50,8 +53,51 @@ func NewModel(ctx *context.ProgramContext) Model {
 	}
 }
 
+func (m *Model) SetAutocomplete(ac *autocomplete.Model) {
+	m.autocomplete = ac
+}
+
 func (m Model) Update(msg tea.Msg) (Model, tea.Cmd) {
 	var cmd tea.Cmd
+
+	switch msg := msg.(type) {
+	case tea.KeyMsg:
+		if m.autocomplete != nil && m.autocomplete.IsVisible() {
+			switch msg.Type {
+			case tea.KeyUp, tea.KeyCtrlP:
+				m.autocomplete.Prev()
+				return m, nil
+			case tea.KeyDown, tea.KeyCtrlN:
+				m.autocomplete.Next()
+				return m, nil
+			case tea.KeyEnter, tea.KeyTab:
+				selected := m.autocomplete.GetSelected()
+				if selected != "" {
+					currentInput := m.textArea.Value()
+					currentLabel := extractCurrentLabel(currentInput)
+					newLabel := selected
+
+					if currentLabel != "" {
+						newInput := strings.ReplaceAll(
```

Comment:
replaceAll here might remove things the user already typed. Let's only remove the current label until the next comma if it exists?